apps/src/applab/ApplabVisualizationColumn.jsx

```diff
@@ -12,32 +23,49 @@ var connect = require('react-redux').connect;
 var ApplabVisualizationColumn = React.createClass({
   propTypes: {
     isReadOnlyWorkspace: React.PropTypes.bool.isRequired,
+    hideSource: React.PropTypes.bool.isRequired,
+    isEmbedView: React.PropTypes.bool.isRequired,
+
     isEditingProject: React.PropTypes.bool.isRequired,
     screenIds: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
     onScreenCreate: React.PropTypes.func.isRequired,
   },
 
   render: function () {
+    var showFrame = !this.props.isShareView;
+
     return (
-      <div id="visualizationColumn">
+      <div
+          id="visualizationColumn"
+          className={classNames({with_padding: this.props.visualizationHasPadding})}
+          style={[
+            (this.props.isEmbedView || this.props.hideSource) && styles.nonResponsive
+          ]}
+      >
         {!this.props.isReadOnlyWorkspace && <PlaySpaceHeader
             isEditingProject={this.props.isEditingProject}
             screenIds={this.props.screenIds}
             onScreenCreate={this.props.onScreenCreate} />
         }
-        <Visualization/>
+        <PhoneFrame showFrame={showFrame} isRunning={this.props.isRunning}>
```

Comment:
This feels like a display-specific component that could live without special knowledge of run-state; so maybe instead of an `isRunning` prop you could just pass it a `darkStyle` prop and let the business logic live here?

Feel free to ignore if you expect upcoming changes to make it more important for `PhoneFrame` to know `isRunning`.